src/librustc_mir/interpret/validity.rs

```diff
@@ -140,6 +141,33 @@ fn scalar_format<Tag>(value: ScalarMaybeUndef<Tag>) -> String {
     }
 }
 
+fn wrapping_range_contains(r: &RangeInclusive<u128>, test: u128) -> bool {
+    let (lo, hi) = r.clone().into_inner();
+    if lo > hi {
+        // Wrapped
+        (..=hi).contains(&test) || (lo..).contains(&test)
+    } else {
+        // Normal
+        r.contains(&test)
+    }
+}
+
+// Formats such that a sentence like "expected something {}" to mean
+// "expected something <in the given range>" makes sense.
+fn wrapping_range_format(r: &RangeInclusive<u128>, max_hi: u128) -> String {
+    let (lo, hi) = r.clone().into_inner();
+    debug_assert!(hi <= max_hi);
+    if lo > hi {
+        format!("less or equal to {}, or greater or equal to {}", hi, lo)
```

Comment:
There are no tests that show this message

Reply:
Any suggestions for how to make one?

Reply:
```rust
#![feature(rustc_attrs)]
#![feature(const_transmute)]

#[rustc_layout_scalar_valid_range_start(10)]
#[rustc_layout_scalar_valid_range_end(30)]
struct Foo(u32);

const F: Foo = unsafe { Foo(42) };
```

Reply:
That will show `in the range 10..=30`, which we probably also didn't test before.

But if I swap the two ranges, I get the error you asked for.